### PR TITLE
Consul updates

### PR DESCRIPTION
**Pull request by HarryMWinters**
- Fixes an issues where the service would error is the registry wasn't reachable.

### CODE REVIEW

File: miteD/middleware/api.py [hunk 1]
```diff
@@ -99,22 +99,35 @@ def _register_with_consul(self):
...
+                except requests.exceptions.RequestException:
+                    logger.info('Error whilst trying to register with consul', exc_info=True)
+                    self.consul_connection_attempts -= 1
+                else:
+                    if r.ok:
+                        self.registered_with_consul = True
+                        # Reset this now that we have registered with consul
+                        # so that we have `RETRY_TTL_PASS` times attempts for
+                        # purely the TTL ping as well
+                        self.consul_connection_attempts = RETRY_TTL_PASS
```
Comment (surajrav) [inline miteD/middleware/api.py]: if possible please rename `RETRY_TTL_PASS` throughout the code to reflect that it is not just a TTL retry but also initial registration retry.

**Review by surajrav — APPROVED** [review]
LGTM... see one minor comment, rest all good, hence, approving.

Also please do check that we did not introduce any regression w.r.t consul and that in the case that the server is actually reachable and that consul still connects and so on...

**Review by pmercado23 — APPROVED** [review]
LGTM :shipit:

**Comment by HarryMWinters** [comment]
Re-named the retry variable and retested yo make sure registration, de-registration and TTL checks were intact.

**Pull request merged**